### PR TITLE
Use ember-cli-release to automate dummy app deploy

### PR DESCRIPTION
**Pull request by offirgolan**
Use ember-cli-release and some random fixes

### CODE REVIEW

File: addon/validations/factory.js [hunk 1]
```diff
@@ -247,7 +247,7 @@ function createCPValidationFor(attribute, validations) {
         let cache = getDebouncedValidationsCacheFor(attribute, model);
         // Return a promise and pass the resolve method to the debounce handler
         value = new Promise(resolve => {
-          cache[getKey(validator)] = run.debounce(validator, () => resolve(validator.validate(attrValue, options, model, attribute)), debounce, false);
+          cache[getKey(validator)] = run.debounce(validator, () => resolve(validator.validate(attrValue, validator.processOptions(), model, attribute)), debounce, false);
```
Comment (offirgolan) [inline addon/validations/factory.js]: There could be times where function based options are time sensitive
and need to be computed right before we call validate. Unfortunately this means having to call `processOptions` twice

File: config/release.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/* jshint node:true */
+var RSVP = require('rsvp');
+var exec = require('child_process').exec;
+
+module.exports = {
+  publish: true,
+  afterPublish: function(project, versions) {
+    // Publish dummy app with docs to gh-pages
+    return runCommand('ember github-pages:commit --message "Released ' + versions.next + '"').then(function() {
```
Comment (rwjblue) [inline config/release.js]: Since you control the system you are running on, I would suggest using `execSync` instead of `exec` here...

```js
var execSync = require('child_process').execSync;

function runCommand(command) {
  console.log('running: ' + command);
  var output = execSync(command, { encoding: 'utf8' });
  console.log(output);
}

module.exports = {
  publish: true,
  afterPublish: function(project, versions) {
    runCommand('ember github-pages:commit --message "Released ' + versions.next + '"');
    runCommand('git push origin gh-pages:gh-pages');
  }
};
```

File: package.json [hunk 3]
```diff
@@ -49,6 +49,7 @@
     "ember-cli-inject-live-reload": "^1.3.0",
     "ember-cli-moment-shim": "0.7.1",
     "ember-cli-qunit": "^1.1.0",
+    "ember-cli-release": "0.2.8",
```
Comment (rwjblue) [inline package.json]: The `publish: true` option was added in https://github.com/lytics/ember-cli-release/pull/39 and is only available in 1.0.0-beta.0.  You may want to update the version here to the beta or call `runCommand('npm publish')` in the `config/release.js`...

File: addon/validations/factory.js (hunk 1 above)
Comment (rwjblue) [inline addon/validations/factory.js]: Not to quibble, but this change seems unrelated to the PR at hand to me.  Am I missing something?

Comment (offirgolan) [inline addon/validations/factory.js]: Hehe yeah my bad, i thought about this super late last night and was too lazy to create a new PR

Comment (offirgolan) [inline addon/validations/factory.js]: Let me move this to a separate PR

**Comment by offirgolan** [comment]
@rwjblue fixed :smile_cat:

**Comment by rwjblue** [comment]
:+1: LGTM

File: addon/validations/factory.js (hunk 1 above)
Comment (stefanpenner) [inline addon/validations/factory.js]: :+1:

File: config/release.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+/* jshint node:true */
+/* var RSVP = require('rsvp'); */
+var execSync = require('child_process').execSync;
+
+module.exports = {
+  publish: true,
+  afterPublish: function(project, versions) {
+    // Publish dummy app with docs to gh-pages
+    runCommand('ember github-pages:commit --message "Released ' + versions.next + '"');
```
Comment (stefanpenner) [inline config/release.js]: these are likely async, i suspect you actually want them to run in series not concurrently.

Comment (rwjblue) [inline config/release.js]: I'm not sure I follow what you mean. These are currently being run in order (not at the same time). See the definition of runCommand below (it's using execSync). Have I misunderstood how execSync works?

**Comment by stefanpenner** [comment]
An. So this requires a native extension to run on node 10?

**Comment by rwjblue** [comment]
@stefanpenner - Yes, but this is only for publishing this addon itself and not for consumers.

**Comment by rwjblue** [comment]
I would be fairly surprised if @offirgolan or yourself are running node 0.10 as your primary node environment for working on addons....

**Comment by stefanpenner** [comment]
@rwjblue well our employer does ;)

File: bower.json [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
-    "ember": "1.13.8",
+    "ember": "1.13.10",
     "ember-cli-shims": "0.0.6",
     "ember-cli-test-loader": "0.2.2",
     "ember-data": "~2.0.0",
     "ember-load-initializers": "0.1.7",
     "ember-qunit-notifications": "0.1.0",
     "ember-resolver": "~0.1.20",
-    "jquery": "^1.11.1",
+    "jquery": "~1.11.3",
```
Comment (rwjblue) [inline bower.json]: Pin to specific version for now. Releases of Ember will be out soon hopefully with the fix (it's already been fixed on master).

**Comment by offirgolan** [comment]
I myself am running node 12 but I have no issue in just wrapping `exec` in a promise.

**Comment by offirgolan** [comment]
@rwjblue I like the idea of using execSync but as a precautionary measure, I think it would be better to just use exec and wrap it in a promise.

**Comment by stefanpenner** [comment]
> I myself am running node 12 but I have no issue in just wrapping exec in a promise.

if its for dev, what exists currently is totally fine :)

**Comment by offirgolan** [comment]
@stefanpenner you guys are killin me here

**Comment by stefanpenner** [comment]
> But if its just for dev, thats fine.

?
